worker/buildbot_worker/base.py

```diff
@@ -320,6 +326,20 @@ def remote_getWorkerInfo(self):
                 if os.path.isfile(filename):
                     with open(filename, "r") as fin:
                         files[f] = fin.read()
+
+        if os.path.exists(self.os_release_file):
```

Comment:
I think a little better way would have been to extract the actual parsing function which gets the contents of the read file or maybe a list of lines. This way we could unit test it directly. Anyway, this is also an acceptable solution.

Reply:
I think this is possibly better way, as @tardyp was concerned about using readlines() earlier.